Modules/_sha3/sha3module.c

```diff
@@ -487,48 +502,24 @@ static PyGetSetDef SHA3_getseters[] = {
     {NULL}  /* Sentinel */
 };
 
+#define SHA3_TYPE_SLOTS(type_slots_obj, type_doc, type_methods) \
+    static PyType_Slot type_slots_obj[] = { \
+        {Py_tp_dealloc, SHA3_dealloc}, \
+        {Py_tp_doc, (char*)type_doc}, \
+        {Py_tp_methods, type_methods}, \
```

Comment:
It seems like SHA3_methods can be hardcoded.

Reply:
@vstinner no actually SHAKE_methods are used by the SHAKE types